docs/content/overriding-styles.mdx

```diff
@@ -0,0 +1,73 @@
+---
+title: Overriding styles with the sx prop
+---
+
+Our goal with Primer Components is to hit the sweet spot between providing too little and too much styling flexibility; too little and the design system is too rigid, and too much and it becomes too difficult to maintain a consistent style. Our components already support a standard set of [system props](/system-props), but sometimes a component just isn't *quite* flexible enough to look the way you need it to look. For those cases, we provide the `sx` prop.
+
+The `sx` prop allows ad-hoc styling that is still theme aware. Declare the styles you want to apply in camel-cased object notation, and any values use appropriate theme values; for example, any `color` property values will use the values from the `colors` key in the theme.
+
+## Basic example
+
+This example demonstrates applying a bottom border to `Heading`, a component that does not receive `BORDER` system props. The `borderBottomWidth` value comes from `theme.borderWidths` and `borderBottomColor` comes from `theme.colors`.
+
+```jsx live
+<Heading pb={2}>Heading</Heading>
+
+<Heading
+  pb={2}
+  sx={{
+    borderBottomWidth: 1,
+    borderBottomColor: 'border.gray',
+    borderBottomStyle: 'solid'
+  }}>
+    Heading with bottom border
+</Heading>
+```
+
+## Responsive values
+
+Just like [values passed to system props](https://styled-system.com/responsive-styles), values in the `sx` prop can be provided as arrays to provide responsive styling.
+
+```jsx live
+<BorderBox
+  p={2}
+  sx={{
+    bg: ['red.1', 'green.1', 'blue.1', 'purple.1', 'yellow.1']
+  }}>
+    Responsive background color
+</BorderBox>
+```
+
+## Nesting, pseudo-classes, and pseudo-elements
+
+The `sx` prop also allows for declaring styles based on media queries, psueudo-classes, and pseudo-elements. This example, though contrived, demonstrates the ability:
+
+```jsx live
+<Box sx={{
+  '> *': {
+    borderWidth: 1,
+    borderColor: 'border.gray',
+    borderStyle: 'solid',
+    borderBottomWidth: 0,
+    padding: 2,
+    ':last-child': {
+      borderBottomWidth: 1
+    },
+    ':hover': {
+      bg: 'gray.1'
+    }
+  }
+}}>
+  <Box>First</Box>
+  <Box>Second</Box>
+  <Box>Third</Box>
+</Box>
+```
+
+## When to use the `sx` prop
+
+The `sx` prop clearly provides a lot of power, which means it is an easy tool to abuse. To best make use of it, we recommend following these guidelines:
+
+* Only use the `sx` prop for small stylistic changes to components. For more substantial changes, create your own wrapper using `styled-components`.
```

Comment:
Is there a reason a component defined with `styled-components`...

```jsx
const MyBox = styled(Box)`
  background: ${get('colors.pink.2')}'
`
```

...is better than a component defined with the `sx` prop?

```jsx
function MyBox(props) {
  return <Box sx={{bg: 'pink.2'}} {...props} />
}
```

I agree that substantial style changes should probably be abstracted into a separate component but I'm curious what advantages the `styled` API provides.

Reply:
Given your two examples, I don't think there is an advantage. Given a large set of styles with pseudo-selectors and the like, I'd say an abstraction like `styled` is probably more readable.

I think my real goal with this bullet point was to suggest patterns that avoid overusing the `sx` prop and providing a huge object of overrides to the built-in components. I'm definitely open to thoughts about the point itself and also the wording tho.